web/main.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:html';
+import 'dart:svg' as svg;
+import 'dart:convert' show JSON;
+
+import 'package:svg_pan_zoom/svg_pan_zoom.dart' as panzoom;
+
+void main() {
+  querySelector('#path-button').onMouseUp.listen(buttonPressHandler);
+
+  var panZoom = new panzoom.SvgPanZoom.selector('.inner');
+  panZoom
+    ..zoomEnabled = true
+    ..panEnabled = true
+    ..zoomSensitivity = 0.02;
+}
+
+void buttonPressHandler(MouseEvent event) {
+  InputElement input = querySelector('#path-to-json');
+  String pathToJson = input.value;
+  HttpRequest.getString(pathToJson).then((String response) {
```

Comment:
Sorry, ignore this, listen wants void, which means that you'd have to do contortions to avoid returning a future.